Fix: Unsupported maccatalyst attribute is getting applied to child attributes but not accounted parent child attributes merge

buyaa-n opened the pull request:
When an API supported on IOS it also supported by maccatalyst as they are related. To remove implicit `maccatalyst` support one need to add `[UnsupportedOSPlatform (""maccatalyst"")]` to the `[SupportedOSPlatform (""ios"")]` 

The `[UnsupportedOSPlatform (""maccatalyst"")]` applied to the immediate (child) attributes list  and removed the implicit `maccatalyst` support  but when child attributes are merged to the parent attributes list this is not accounted and merged list will keep the implicit `maccatalyst` support. The PR fixes this bug


Fixes https://github.com/dotnet/roslyn-analyzers/issues/7633

rolfbjarne commented:
This regresses the following sample/unit test: https://github.com/rolfbjarne/roslyn-analyzers/commit/84ddcbc30470d73378084d0274d7ad062eed9a88, it shows:

```
Diagnostics:
// /0/Test0.cs(22,20): warning CA1416: This call site is reachable on: 'ios', 'macOS/OSX', 'tvos'. 'NativeObjectExtensions.GetHandle(INativeObject)' is only supported on: 'maccatalyst'.
```

buyaa-n commented:
> This regresses the following sample/unit test: [rolfbjarne@84ddcbc](https://github.com/rolfbjarne/roslyn-analyzers/commit/84ddcbc30470d73378084d0274d7ad062eed9a88), it shows:
> 
> ```
> Diagnostics:
> // /0/Test0.cs(22,20): warning CA1416: This call site is reachable on: 'ios', 'macOS/OSX', 'tvos'. 'NativeObjectExtensions.GetHandle(INativeObject)' is only supported on: 'maccatalyst'.
> ```

Thanks, when implicit `maccatalyst` support added with [SupportedOSPlatform (""ios"")] is suppressed by `[UnsupportedOSPlatform (""maccatalyst"")]` current logic removes `maccatalyst` support version but keeps the platform in the list with empty (null) versions. This empty `maccatalyst` causing this regression, it shows that we should remove this platform completely. By design child (nested) API could narrow platform support, removing empty `maccatalyst` will just work as pre this scenario. I have updated the code accordingly and added related tests

rolfbjarne commented:
CC @ViktorHofer @tannergooding, is this something you guys can approve/merge? I (as the bug reporter) can verify the fix works, I've tested it locally.
